src/main/java/org/commcare/formplayer/session/MenuSession.java

```diff
@@ -303,12 +302,14 @@ private void computeDatum() {
 
     public FormSession getFormEntrySession(FormSendCalloutHandler formSendCalloutHandler,
                                            FormplayerStorageFactory storageFactory,
-                                           CaseSearchHelper caseSearchHelper) throws Exception {
+                                           CaseSearchHelper caseSearchHelper,
+                                           FormDefinitionService formDefinitionService) throws Exception {
         String formXmlns = sessionWrapper.getForm();
-        FormDef formDef = engine.loadFormByXmlns(formXmlns);
+        String serializedFormDef = FormDefStringSerializer.serialize(this.engine.loadFormByXmlns(formXmlns));
```

Comment:
same comment here about moving the serialisation into the service

Reply:
See ed0229dc5755f8fb95512c5a60640cafa43ea457